### ISSUE TITLE
Cannot convert undefined or null to object Exiting

### ISSUE THREAD

**Issue by agnosticcloud** [IBM/audit-ci#247]
This is the output of the audit-ci
```
audit-ci --config=audit-ci.json 2$>audit-report.json

[31mCannot convert undefined or null to object[0m
[31mExiting...[0m
```
These are the versions
```
Audit-ci: 5.1.2
NPM: 8.5.0
Nodejs: v16.14.1
```
Can you please confirm if the NPM/Nodejs versions are compatible with the audit-ci version? Let me know if you need any additional details. Thanks in advance.

**Comment by quinnturner** [comment]
Do you have a lockfile?

Also, please consider updating to the latest `audit-ci`, as the dependency resolution approach changed for the better!

**Comment by agnosticcloud** [comment]
Yes we have a lockfile, and we are also running into this error when using the latest version of audit-ci.

**Comment by quinnturner** [comment]
Do you have private dependencies?

Is it possible for you to email me the `package.json` and lockfile? You can remove the name of the project; I'd like a reproduction to help understand where the issue occurs. quinnDOTturnerATuwaterlooDOTca

**Comment by 1sherlynn** [comment]
Are there any updates/findings on this? Also facing a similar issue. 

```
audit-ci version: 6.0.0
Cannot convert undefined or null to object
Exiting...
```

Versions: 
```
Audit-ci: 6.2.0 (latest at the moment)
NPM: 8.1.2
Nodejs: v16.13.2
```

**Comment by quinnturner** [comment]
Thank you @1sherlynn for the report! Good to know that it's still occurring and that it's on NPM 8. Hoping you can post the result of this command: npm audit --json

**Comment by aarbro** [comment]
We were having the same issue after upgrading to node 16 and audit-ci 6.2.x. 

I just figured out that we had a private registry in the project's .npmrc file. Once I overrode the registry to the default via the audit-ci command line (e.g. `audit-ci --registry https://registry.npmjs.org`), I no longer got the error. Setting it in the audit-ci config file worked also.

You can reproduce the error by simply specifying a bogus registry (e.g. `audit-ci --registry http://localhost`).

It would be great to get a fix for this, as it has caused lots of confusion!

**Comment by quinnturner** [comment]
Thanks for the heads up @aarbro, will look into it and mark this as a bug!

**Comment by quinnturner** [comment]
PR to resolve the registry issue @aarbro mentioned (and might catch other NPM 7+ related issues): https://github.com/IBM/audit-ci/pull/258

**Comment by quinnturner** [comment]
Released in `6.2.1`. If you're still having this issue, please message!
